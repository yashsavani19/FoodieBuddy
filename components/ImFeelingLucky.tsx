import { Pressable, StyleSheet, Text, Touchable, View } from "react-native";
import React, { useContext } from "react";
import { AppContext } from "@/context/AppContext";
import { useNavigation } from "expo-router";
import { RootStackParamList } from "@/constants/navigationTypes";
import { NavigationProp } from "@react-navigation/native";

// props: children components
// return: Pressable component
interface ImFeelingLuckyProps {
  children: React.ReactNode;
}

const ImFeelingLucky: React.FC<ImFeelingLuckyProps> = ({ children }) => {
  const { localRestaurants } = useContext(AppContext);
  const navigation = useNavigation<NavigationProp<RootStackParamList>>();

  const generateRandomNumber = () => {
    return Math.floor(Math.random() * localRestaurants.length);
  };

  const handleRandom = () => {
    const randomIndex = generateRandomNumber();
    console.log(`Random index: ${randomIndex}`);
    console.log(`Random restaurant: ${localRestaurants[randomIndex].name}`);
    navigation.navigate("DetailsView", {
      Restaurant: localRestaurants[randomIndex],
    });
  };

  return (
    <View>
      <Pressable testID="lucky-button" onPress={handleRandom}>{children}</Pressable>
    </View>
  );
};

export default ImFeelingLucky;

const styles = StyleSheet.create({});
