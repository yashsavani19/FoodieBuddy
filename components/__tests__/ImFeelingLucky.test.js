import { render, fireEvent } from "@testing-library/react-native";
import ImFeelingLucky from "../ImFeelingLucky";

// describe("ImFeelingLucky Component", () => {
//   it("Button renders correctly", () => {
//     const tree = render(<ImFeelingLucky />).toJSON();
//     expect(tree).toBeOnTheScreen();
//   });

//   it("renders the correct style", () => {
//     const { getByTestId } = render(<ImFeelingLucky />);
//     expect(getByTestId("lucky-button").props.style).toEqual({
//       // styles to be tested
//     });
//   });
// });

// Test to see if a random restaurant is selected
test("ImFeelingLucky button press", () => {
  const { getByTestId } = render(<ImFeelingLucky />);
  fireEvent.press(getByTestId("lucky-button"));

  // Check if a restaurant is selected (returned restaurant not null)
  expect(getByTestId("selected-restaurant")).toBeTruthy();
});
