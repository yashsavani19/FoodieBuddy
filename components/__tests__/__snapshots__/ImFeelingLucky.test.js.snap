// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImFeelingLucky Component Button renders correctly 1`] = `
<View>
  <Text>
    ImFeelingLucky
  </Text>
</View>
`;
